src/test/HostActivationTests/DotNetBuilder.cs

```diff
@@ -65,6 +65,66 @@ public DotNetBuilder AddMicrosoftNETCoreAppFramework(string version)
             return this;
         }
 
+        /// <summary>
+        /// Add a mock of the Microsoft.NETCore.App framework with the specified version
+        /// </summary>
+        /// <param name="version">Version to add</param>
+        /// <remarks>
+        /// Product runtime binaries are not added. All the added mock framework will contain is hostpolicy,
+        /// a mock version of coreclr, and a minimal Microsoft.NETCore.App.deps.json.
+        /// </remarks>
+        public DotNetBuilder AddMicrosoftNETCoreAppFrameworkMockCoreClr(string version)
+        {
+            // ./shared/Microsoft.NETCore.App/<version> - create a mock of the root framework
+            string netCoreAppPath = Path.Combine(_path, "shared", "Microsoft.NETCore.App", version);
+            Directory.CreateDirectory(netCoreAppPath);
+
+            // ./shared/Microsoft.NETCore.App/<version>/hostpolicy.dll - this is the real component and will load CoreClr library
+            string hostPolicyFileName = RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("hostpolicy");
+            File.Copy(
+                Path.Combine(_repoDirectories.Artifacts, "corehost", hostPolicyFileName),
+                Path.Combine(netCoreAppPath, hostPolicyFileName),
+                true);
+
+            // ./shared/Microsoft.NETCore.App/<version>/coreclr.dll - this is a mock, will not actually run CoreClr
+            string coreclrFileName = RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("coreclr");
+            string mockCoreclrFileName = RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("mockcoreclr");
+            File.Copy(
+                Path.Combine(_repoDirectories.Artifacts, "corehost_test", mockCoreclrFileName),
+                Path.Combine(netCoreAppPath, coreclrFileName),
+                true);
+
+            string netCoreAppPathDepsJson = Path.Combine(netCoreAppPath, "Microsoft.NETCore.App.deps.json");
+
+            string depsJsonBody = $@"{{
+              ""runtimeTarget"": "".NETCoreApp"",
+              ""targets"": {{
+                "".NETCoreApp"": {{
+                  ""Microsoft.NETCore.App/{version}"": {{
+                    ""native"": {{
+                      ""runtimes/any/native/{coreclrFileName}"": {{ }}
+                    }}
+                  }},
+                  ""runtime.any.Microsoft.NETCore.DotNetHostPolicy/{version}"": {{
+                    ""native"": {{
+                      ""runtimes/any/native/{hostPolicyFileName}"": {{}}
+                    }}
+                  }}
+                }}
+              }},
+              ""libraries"": {{
+                ""Microsoft.NETCore.App/{version}"": {{
+                  ""type"": """",
```

Comment:
I think `type` should be `package` - at least that's what's in the .deps.json in the real NETCoreApp.

Reply:
I was stripping to get a minimal deps.json.  I removed it since the hostpolicy parser was ignoring the value. I'll add it back.

Reply:
I haven't looked at `hostpolicy` just at the frameworks as we have them today... I'm fine if you want to keep it empty.